api/src/main/java/jakarta/persistence/criteria/ComparableExpression.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2008, 2023 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Christian Beikov - 3.2
+
+
+package jakarta.persistence.criteria;
+
+/**
+ * Type for comparable query expressions.
+ *
+ * @param <X> the type of the expression
+ *
+ * @since 3.2
+ */
+@SuppressWarnings("hiding")
+public interface ComparableExpression<X extends Comparable<? super X>> extends Expression<X> {
```

Comment:
Should add `Order asc()` and `Order desc()` here, right? And possibly overloads accepting `Nulls`.